src/ebonite/client/base.py

```diff
@@ -97,11 +109,22 @@ def get_or_create_task(self, project_name: str, task_name: str) -> Task:
         task.bind_artifact_repo(self.artifact_repo)
         return task
 
+    def delete_task(self, task, *, cascade=False):
+        """
+        Deletes task and(if required) all models associated with it
+        :param task: task which is meant to be deleted
+        :param cascade: whether should task be deleted with all associated models
+        :return: Nothing
+        """
+        if cascade:
+            for model in task.models:
```

Comment:
I recommend using `self.meta_repo.get_models` here as `Task.models` is not guaranteed to reflect DB state.